pkg/resourceutil/sort_test.go

```diff
@@ -15,33 +15,99 @@
 package resourceutil
 
 import (
+	"fmt"
+	"reflect"
 	"testing"
 
-	"github.com/stretchr/testify/require"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 func TestSortResourcesByKind(t *testing.T) {
-	resources := []Resource{
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Secret"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRole"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "IngressRoute"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRoleBinding"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "PodSecurityPolicy"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ServiceAccount"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Service"}},
-	}
-
 	t.Run("Reordering resources based on default reordering function", func(t *testing.T) {
-		expected := []string{"PodSecurityPolicy", "ServiceAccount", "ClusterRole", "ClusterRoleBinding", "Secret", "ConfigMap", "ConfigMap", "Service", "Deployment", "IngressRoute", "Unknown"}
+		resources := []Resource{
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Secret"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRole"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "IngressRoute"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRoleBinding"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "PodSecurityPolicy"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ServiceAccount"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Service"}},
```

Comment:
Can we add the SecretProviderClass test case?

Reply:
Done